ee/maintained-apps/inputs/darwin/ingester.go

```diff
@@ -0,0 +1,370 @@
+	}
+
+	out.Version = cask.Version
+	out.InstallerURL = cask.URL
+	out.UniqueIdentifier = app.UniqueIdentifier
+	out.SHA256 = cask.SHA256
+	out.Queries = map[string]string{maintained_apps.QueryKeyExists: fmt.Sprintf("SELECT 1 FROM apps WHERE bundle_identifier = '%s';", out.UniqueIdentifier)}
+	out.Description = cask.Desc
+	out.Slug = fmt.Sprintf("%s/%s", cask.Token, fleet.MacOSPlatform)
+
+	// Script generation
+	scriptRefs := make(map[string]string)
+	uninstallScript := uninstallScriptForApp(&cask)
+	uninstallRef := uuid.NewString()
+	out.UninstallScriptRef = uninstallRef
+	installScript, err := installScriptForApp(app, &cask)
+	if err != nil {
+		return nil, nil, ctxerr.WrapWithData(ctx, err, "generating install script for maintained app", map[string]any{"unique_identifier": app.UniqueIdentifier})
+	}
+	installRef := uuid.NewString()
+	scriptRefs[installRef] = installScript
+	out.InstallScriptRef = installRef
+
+	scriptRefs[uninstallRef] = uninstallScript
+
+	return out, scriptRefs, nil
+}
+
+func shouldUpdateApp(ctx context.Context, sourceID string, outApp *maintained_apps.FMAManifestApp) (bool, error) {
+	existingFileBytes, err := os.ReadFile(path.Join(outputPath, string(fleet.MacOSPlatform), fmt.Sprintf("%s.json", sourceID)))
+	if err != nil {
+		if err == os.ErrNotExist {
+			return true, nil
+		}
+
+		return false, ctxerr.Wrap(ctx, err, "reading existing app manifest")
+	}
+
+	var existingManifest maintained_apps.FMAManifestFile
+	if err := json.Unmarshal(existingFileBytes, &existingManifest); err != nil {
+		return false, ctxerr.Wrap(ctx, err, "unmarshaling existing app manifest", "slug", outApp.Slug)
+	}
+
+	if len(existingManifest.Versions) < 1 {
+		return false, ctxerr.Wrap(ctx, err, "invalid existing app manifest", "slug", outApp.Slug)
+	}
+
+	if existingManifest.Versions[0].SHA256 == outApp.SHA256 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func updateAppsListFile(ctx context.Context, input maintained_apps.InputApp, outApp *maintained_apps.FMAManifestApp) error {
+	appListFilePath := path.Join(outputPath, "apps.json")
+	file, err := os.ReadFile(appListFilePath)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "reading output apps list file")
+	}
+
+	var outputAppsFile maintained_apps.FMAListFile
+	if err := json.Unmarshal(file, &outputAppsFile); err != nil {
+		return ctxerr.Wrap(ctx, err, "unmarshaling output apps list file")
+	}
+
+	var found bool
+	for _, a := range outputAppsFile.Apps {
+		if a.Slug == fmt.Sprintf("%s/%s", input.SourceIdentifier, fleet.MacOSPlatform) {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		outputAppsFile.Apps = append(outputAppsFile.Apps, maintained_apps.FMAListFileApp{
+			Name:             input.Name,
+			Slug:             fmt.Sprintf("%s/%s", input.SourceIdentifier, fleet.MacOSPlatform),
+			Platform:         string(fleet.MacOSPlatform),
+			UniqueIdentifier: outApp.UniqueIdentifier,
+			Description:      outApp.Description,
```

Comment:
Let's drop description out of the input format etc.; people can edit the apps.json file directly to add description, and dropping it from here means there's eactly one place to edit vs. it being less clear why, if a description is updated in one place, it doesn't get updated in the other (because we don't update entries in apps.json if they already exist).